packages/web/src/components/address-tile/AddressTile.tsx

```diff
@@ -1,49 +1,102 @@
-import { ReactNode, useCallback, useContext } from 'react'
+import { useCallback, useContext } from 'react'
 
-import { shortenSPLAddress } from '@audius/common'
-import { Text, IconCopy, PlainButton } from '@audius/harmony'
+import {
+  BNUSDC,
+  decimalIntegerToHumanReadable,
+  formatUSDCWeiToCeilingCentsNumber,
+  shortenSPLAddress,
+  useUSDCBalance
+} from '@audius/common'
+import {
+  Text,
+  IconCopy,
+  PlainButton,
+  IconComponent,
+  Flex,
+  Box
+} from '@audius/harmony'
+import { BN } from 'bn.js'
 
-import { Icon } from 'components/Icon'
 import { ToastContext } from 'components/toast/ToastContext'
 import { isMobile } from 'utils/clientUtil'
 import { copyToClipboard } from 'utils/clipboardUtil'
 
-import styles from './AddressTile.module.css'
-
 const messages = {
+  usdcBalance: 'USDC Balance',
   copied: 'Copied to Clipboard!'
 }
 
 type AddressTileProps = {
   address: string
-  left?: ReactNode
-  right?: ReactNode
+  iconLeft: IconComponent
+  iconRight?: IconComponent
 }
 
-export const AddressTile = ({ address, left, right }: AddressTileProps) => {
+export const AddressTile = ({
+  address,
+  iconLeft: IconLeft,
+  iconRight: IconRight
+}: AddressTileProps) => {
   const { toast } = useContext(ToastContext)
   const mobile = isMobile()
 
+  const { data: balance } = useUSDCBalance({ isPolling: true })
+  const balanceNumber = formatUSDCWeiToCeilingCentsNumber(
+    (balance ?? new BN(0)) as BNUSDC
+  )
+  const balanceFormatted = decimalIntegerToHumanReadable(balanceNumber)
+
   const handleCopyPress = useCallback(() => {
     copyToClipboard(address)
     toast(messages.copied)
   }, [address, toast])
 
   const defaultRight = (
-    <PlainButton>
-      <Icon icon={IconCopy} onClick={handleCopyPress} />
+    <PlainButton onClick={handleCopyPress}>
+      <IconCopy size='s' color='subdued' />
     </PlainButton>
   )
 
   return (
-    <div className={styles.addressContainer}>
-      <div className={styles.leftContainer}>{left}</div>
-      <div className={styles.middleContainer}>
-        <Text variant='body' className={styles.address}>
-          {shortenSPLAddress(address, mobile ? 6 : 12)}
+    <Flex direction='column' border='default' borderRadius='s'>
+      <Flex p='l' alignItems='center' justifyContent='space-between'>
+        <Flex alignItems='center'>
+          <IconLeft />
+          <Box pl='s'>
+            <Text variant='title' size='m'>
+              {messages.usdcBalance}
+            </Text>
+          </Box>
+        </Flex>
+        <Text variant='title' size='l' strength='strong'>
```

Comment:
Is it `size='l'` now? I thought these were full word values on `Text`

Reply:
it's definitely size='l' now... not sure???